File: leetcode/leetcode_c/maxSubArray.cc

#include<vector>
#include<algorithm>

class Solution{
    public:
        int maxSubAttay(vector<int>& nums){
            int pre = 0,maxAns = nums[0];
            for (auto &&x : nums)
            {
                pre = max(pre+x,x);
                maxAns = max(maxAns,pre);
            }
            return maxAns;
        }
}

File: leetcode/leetcode_java/src/dp21/maxSubArray.java
package dp21;

/**
 * @ClassName: maxSubArray
 * @Description: dp 03
 * @Author: jackey
 * @Create: 2021/8/29 下午9:20
 */
class Solution {
    public int maxSubArray(int[] nums) {
        int max = nums[0], curSum = 0;
        for (int x : nums) {
            curSum = Math.max(curSum + x, x);
            max = Math.max(curSum, max);
        }
        return max;
    }
}
